sgx-types/src/ssa.rs

```diff
@@ -0,0 +1,135 @@
+#![allow(clippy::unreadable_literal)]
+
+#[derive(Copy, Clone, Debug)]
+pub enum Exception {
+    Divider,
+    Debug,
+    Breakpoint,
+    BoundRange,
+    InvalidOpCode,
+    GeneralProtection,
+    PageFault,
+    FloatingPoint,
+    AlignmentCheck,
+    Simd,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum ExitType {
+    Hardware,
+    Software,
+}
+
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ExitInfo(u32);
+
+impl std::fmt::Debug for ExitInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let exc = self.exception();
+        let et = self.exit_type();
+
+        let status = et.and_then(|et| exc.map(|e| (et, e)));
+        write!(f, "ExitInfo({:?})", status)
+    }
+}
+
+impl ExitInfo {
+    pub fn new(et: ExitType, exc: Exception) -> Self {
+        let et = match et {
+            ExitType::Hardware => 0b011 << 8,
+            ExitType::Software => 0b110 << 8,
+        };
+
+        let exc = match exc {
+            Exception::Divider => 0x00,
```

Comment:
Where are these values from?

Reply:
They are the Intel CPU exception constants. I found a list in the SGX docs. But I'd like to improve this section generally.